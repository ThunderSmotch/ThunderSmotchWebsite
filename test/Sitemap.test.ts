describe("Sitemap", () => {
    let Sitemap: typeof import("../src/Sitemap");

    beforeEach(() => {
        return import('../src/Sitemap').then(module => {
            Sitemap = module;
            jest.resetModules();
        })
    })

    it("should not be started twice", () => {
        expect(Sitemap.Start).not.toThrow(Error);
        expect(Sitemap.Start).toThrow(Error);
    });

    it("should not be ended twice", () => {
        expect(Sitemap.Start).not.toThrow(Error);
        expect(Sitemap.End).not.toThrow(Error);
        expect(Sitemap.End).toThrow(Error);
    });

    it("should not be started after ending", () => {
        Sitemap.Start();
        Sitemap.End();
        expect(Sitemap.Start).toThrow(Error);
    });

    it("should not end before start is called", () => {
        expect(Sitemap.End).toThrow(Error);
        Sitemap.Start();
        expect(Sitemap.End).not.toThrow(Error);
    });

    it("should not output XML if imcomplete", () => {
        expect(Sitemap.GetXML).toThrow();
        Sitemap.Start();
        expect(Sitemap.GetXML).toThrow();
        Sitemap.End();
        expect(Sitemap.GetXML).not.toThrow();
    });

    it("should not allow adding empty urls", () => {
        Sitemap.Start();
        expect(()=>Sitemap.AddURL("")).toThrow();
    });

    it("outputs correct basic sitemap if no urls are added", () => {
        Sitemap.Start();
        Sitemap.End();
        expect(Sitemap.GetXML).not.toThrow();
        expect(Sitemap.GetXML()).toMatchSnapshot();
    });

    it("outputs correct sitemap when adding urls", () => {
        Sitemap.Start();
        Sitemap.AddURL("https://thundersmotch.com/test/");
        Sitemap.End();
        expect(Sitemap.GetXML()).toMatchSnapshot();
    });

});