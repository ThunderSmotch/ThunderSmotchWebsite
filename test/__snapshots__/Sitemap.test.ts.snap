// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sitemap outputs correct basic sitemap if no urls are added 1`] = `
"<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
</urlset>"
`;

exports[`Sitemap outputs correct sitemap when adding urls 1`] = `
"<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
<url>
    <loc>https://thundersmotch.com/test/</loc>
    <changefreq>daily</changefreq>
</url>
</urlset>"
`;
